src/plugins/typechecker/README.md

```diff
@@ -0,0 +1,115 @@
+- infos = Information about the typechecker plugin is in keys below
+- infos/author = e1528532 <e1528532@libelektra.org>
+- infos/licence = BSD
+- infos/needs = 
+- infos/provides = typechecker
+- infos/recommends =
+- infos/placements = postgetstorage presetstorage
+- infos/status = maintained experimental global
+- infos/metadata =
+- infos/description = a plugin which typechecks configuration specifications
+
+## Introduction
+
+A plugin which typechecks specifications before setting keys and after getting keys from
+a mounted specification.
+
+The typesystem is currently based on regular expressions. Each key is assigned with a regex
+that describes its contents. Links between keys can only be done if the regexes describing 
+the linked keys are compatible with each other.
+
+## Usage
+
+In order to use the type checker, mount a configuration specification along with this
+plugin. 
+
+`kdb mount <specification> spec/<path> <storage plugin to read the specification> typechecker`
+
+Currently there is no way to check the specification automatically upon mounting. Simply 
```

Comment:
Does not need to be written here (already is down in the limitations)

Reply:
fixed